src/Microsoft.Net.Http.Headers/MediaTypeHeaderValue.cs

```diff
@@ -0,0 +1,408 @@
+            }
+        }
+
+        /// <summary>
+        /// MediaType = "*/*"
+        /// </summary>
+        public bool AllTypes
+        {
+            get
+            {
+                return MediaType.Equals("*/*", StringComparison.Ordinal);
+            }
+        }
+
+        /// <summary>
+        /// SubType = "*"
+        /// </summary>
+        public bool AllSubTypes
+        {
+            get
+            {
+                return SubType.Equals("*", StringComparison.Ordinal);
+            }
+        }
+
+        public bool IsSubsetOf(MediaTypeHeaderValue otherMediaType)
+        {
+            if (otherMediaType == null)
+            {
+                return false;
+            }
+
+            if (!Type.Equals(otherMediaType.Type, StringComparison.OrdinalIgnoreCase))
+            {
+                if (!otherMediaType.AllTypes)
+                {
+                    return false;
+                }
+            }
+            else if (!SubType.Equals(otherMediaType.SubType, StringComparison.OrdinalIgnoreCase))
+            {
+                if (!otherMediaType.AllSubTypes)
+                {
+                    return false;
+                }
+            }
+
+            if (Parameters != null)
+            {
+                if (Parameters.Count != 0 && (otherMediaType.Parameters == null || otherMediaType.Parameters.Count == 0))
+                {
+                    return false;
+                }
+
+                // Make sure all parameters listed locally are listed in the other one. The other one may have additional parameters.
+                foreach (var param in _parameters)
+                {
+                    var otherParam = NameValueHeaderValue.Find(otherMediaType._parameters, param.Name);
+                    if (otherParam == null)
+                    {
+                        return false;
+                    }
+                    if (!string.Equals(param.Value, otherParam.Value, StringComparison.OrdinalIgnoreCase))
+                    {
+                        return false;
+                    }
+                }
+            }
+
+            return true;
+        }
+
+        public override string ToString()
+        {
+            return _mediaType + NameValueHeaderValue.ToString(_parameters, ';', true);
+        }
+
+        public override bool Equals(object obj)
+        {
+            var other = obj as MediaTypeHeaderValue;
+
+            if (other == null)
+            {
+                return false;
+            }
+
+            return (string.Compare(_mediaType, other._mediaType, StringComparison.OrdinalIgnoreCase) == 0) &&
+                HeaderUtilities.AreEqualCollections(_parameters, other._parameters);
+        }
+
+        public override int GetHashCode()
+        {
+            // The media-type string is case-insensitive.
+            return _mediaType.ToLowerInvariant().GetHashCode() ^ NameValueHeaderValue.GetHashCode(_parameters);
```

Comment:
Use StringComparer.OrdinalIgnoreCase.GetHashCode(string) instead.  It's closer to what you actually want and saves an allocation.